internal/eval/eval.go

```diff
@@ -4108,6 +4110,91 @@ func evalPFMERGE(args []string, store *dstore.Store) []byte {
 	return clientio.RespOK
 }
 
+func evalPFDEBUG(args []string, store *dstore.Store) []byte {
+	if len(args) != 2 {
+		return diceerrors.NewErrArity("PFDEBUG")
+	}
+
+	subcommand := strings.ToUpper(args[0])
+	key := args[1]
+
+	obj := store.Get(key)
+	if obj == nil {
+		return diceerrors.NewErrWithMessage("Key does not exist")
+	}
+
+	hll, ok := obj.Value.(*hyperloglog.Sketch)
+	if !ok {
+		return diceerrors.NewErrWithMessage(diceerrors.WrongTypeHllErr)
+	}
+
+	sparseField := reflect.ValueOf(hll).Elem().FieldByName("sparseList")
+	sparseFieldValue := reflect.NewAt(sparseField.Type(), unsafe.Pointer(sparseField.UnsafeAddr())).Elem()
```

Comment:
Is there a way we can avoid usage of unsafe pointers?

Reply:
one thing we can do, convert the hyperloglog structure to binary, the third byte is 0/1 depending on sparse/dense notation. I'll use that instead.